assets/app/view/game/buy_trains.rb

```diff
@@ -16,56 +16,61 @@ class BuyTrains < Snabberb::Component
 
       def render_president_contributions
         player = @corporation.owner
+        step = @game.round.active_step
 
         children = []
 
         verb = @must_buy_train ? 'must' : 'may'
 
+        cheapest_train_price = if step.respond_to?(:cheapest_train_price)
+                                 step.cheapest_train_price(@corporation)
+                               else
+                                 @depot.min_depot_price
+                               end
         cash = @corporation.cash + player.cash
-        share_funds_required = @depot.min_depot_price - cash
+        share_funds_required = cheapest_train_price - cash
         share_funds_allowed = if @game.class::EBUY_DEPOT_TRAIN_MUST_BE_CHEAPEST
                                 share_funds_required
```

Comment:
Instead of having multiple conditionals that check `@game.class::EBUY_CAN_SELL_SHARES`, maybe `share_funds_allowed` could just be set to 0? Something like this:

```suggestion
        share_funds_allowed = if @game.class::EBUY_CAN_SELL_SHARES
                                0
                              elsif @game.class::EBUY_DEPOT_TRAIN_MUST_BE_CHEAPEST
                                share_funds_required
```

Reply:
When I just set share_funds_allowed to 0 , 
```
 if share_funds_required.positive?
            children << h(:div, "#{player.name} #{verb} sell shares to raise at least "\
                                "#{@game.format_currency(share_funds_required)}.")
          end
 ```
 still is true. I made an big if around all of the texts